### ISSUE TITLE
[Request]: In-app server about page

### ISSUE THREAD

**Issue by TriVoxel** [GeopJr/Tuba#810]
## Description

It would be nice if users could check the server's [about page](https://mastodon.social/about) in-app.

Most servers have a description, rules, links to moderators, etc. I'd like it if I could see this information in-app, for instance, to message the current moderators to report something, see updated rules, and just check in on the state of the server. May be useful for newer instances, or for server admins to check that their changes are updated. It's always good when users can easily find their server's rules somewhere in-app.

## Design

Perhaps the client could parse the HTML from the website, and this functionality could be added in the form of a button in the main menu called "About Server" which could be right under "Open profile". It could open a simple modal window with the about page parsed into GTK4, with clickable links, and moderators listed can be clicked on to close the window and redirect the user to their respective profile.

### Implementation Details

- [X] This follows the [GNOME HIG](https://developer.gnome.org/hig/).

**Comment by GeopJr** [comment]
Sounds good to me! But I'd probably place it in settings instead to avoid wasting a menu item for something most users will only view once (at best)

I think Megalodon does it too, along with other links or info

**Comment by TriVoxel** [comment]
> Sounds good to me! But I'd probably place it in settings instead to avoid wasting a menu item for something most users will only view once (at best)

It isn't really a setting per se, though. I don't feel like that would be a very logical or intuitive place tbh. Perhaps in the "About" menu, there could be a "My host" with the host's URL beneath it in small text?

The reason I said in the main menu is because it makes it easy to find, could be self-contained, and since opening your profile (on the server) can be done there, It seemed like a logical place to have an "About server" or "Rules" button there.

Additionally, I think it would be cool if, upon linking an account, this "about server" page appeared so that the user could review the server rules before using it, in the event that they just signed up for an account and immediately went to use Tuba, it would ensure they know the rules, and would only take one click to dismiss it.

**Comment by GeopJr** [comment]
I agree that placing it in settings doesn't make much sense on its own, but along with other links it kind-of does? It could be a small section wtih "About \<instance name\>", "Security Settings" and other settings we don't plan to handle... On a second thought however, this would be extremely messy on non-mastodon backends as they have different endpoitns :shrug: 

The about window is provided as is. We could maybe do a workaround and provide a custom tuba:// link we handle to open it but instance info is unrelated to "About Tuba" so probably shouldn't.

I guess the main menu is the only option

> Additionally, I think it would be cool if, upon linking an account, this "about server" page appeared so that the user could review the server rules before using it, in the event that they just signed up for an account and immediately went to use Tuba, it would ensure they know the rules, and would only take one click to dismiss it.

Not opposed to it but when creating an account Mastodon makes sure you read and agreed to the rules. In the future we might just add a way to register though (to be honest, I *dont* want Tuba to be responsible for e-mails and passwords but some backends like gotosocial don't have a UI for registering)

megalodon screenshot for reference:
![image](https://github.com/GeopJr/Tuba/assets/18014039/95ac5eea-ff61-4db9-8552-ca137453ac21)
